File: Web/src/lib/client/Players/PlayerExistsInGame.ts
import { createEndpoint } from "../api";
export type PlayerExistsInGameRequest = {
    readonly id: string;
    readonly gameId: string;
};
export type PlayerExistsInGameResponse = {
    readonly exists: boolean;
};
export const PlayerExistsInGame = createEndpoint<PlayerExistsInGameRequest, PlayerExistsInGameResponse>('/api/players/{id}/{gameId}/exists', 'GET');

File: API/Features/Players/Endpoints/PlayerExistsInGame.cs
using API.DataAccess.Repositories;
using API.Domain;

namespace API.Features.Players.Endpoints;

public static class PlayerExistsInGame
{
    public record Response(bool Exists);
    public static async Task<IResult> HandleAsync(PlayerRepository repository, int playerId, int gameId)
    {
        var player = await repository.GetAsync(playerId);

        if(player == null)
        {
            return Results.Ok(new Response(false));
        }
        if(player.GameId != gameId)
        {
            return Results.Ok(new Response(false));
        }

        return Results.Ok(new Response(true));
    }
}
